test/test_query.cpp

```diff
@@ -1816,6 +1816,140 @@ TEST(Query_StrIndexCrash)
     }
 }
 
+TEST(Query_size)
+{
+    Group g;
+
+    TableRef table1 = g.add_table("primary");
+    TableRef table2 = g.add_table("secondary");
+    TableRef table3 = g.add_table("top");
+
+    table1->add_column(type_String, "strings");
+    table1->add_column(type_Binary, "binaries", true);
+    DescriptorRef subdesc;
+    table1->add_column(type_Table, "intlist", false, &subdesc);
+    subdesc->add_column(type_Int, "list", nullptr, true);
+    table1->add_column_link(type_LinkList, "linklist", *table2);
+
+    table2->add_column(type_Int, "integers");
+
+    table3->add_column_link(type_Link, "link", *table1);
+    table3->add_column_link(type_LinkList, "linklist", *table1);
+    table3->add_empty_row(10);
+
+    Columns<String> strings = table1->column<String>(0);
+    Columns<Binary> binaries = table1->column<Binary>(1);
+    Columns<SubTable> intlist = table1->column<SubTable>(2);
+    Columns<LinkList> linklist = table1->column<LinkList>(3);
+
+    table1->add_empty_row(10);
+    table2->add_empty_row(10);
+
+    for (size_t i = 0; i < 10; i++) {
+        table2->set_int(0, i, i);
+    }
+
+    // Leave the last one null
+    for (unsigned i = 0; i < 9; i++) {
+        table3->set_link(0, i, i % 4);
+    }
+
+    for (unsigned i = 0; i < 10; i++) {
+        auto lv = table3->get_linklist(1, i);
+        for (unsigned j = 0; j < i % 5; j++) {
+            lv->add(j);
+        }
+    }
+
+    table1->set_string(0, 0, StringData("Hi"));
+    table1->set_string(0, 1, StringData("world"));
+
+    std::string bin1(100, 'a');
+    std::string bin2(500, '5');
+    table1->set_binary(1, 0, BinaryData(bin1));
+    table1->set_binary(1, 1, BinaryData(bin2));
+
+    auto set_list = [](TableRef subtable, const std::vector<int64_t>& value_list) {
+        size_t sz = value_list.size();
+        subtable->clear();
+        subtable->add_empty_row(sz);
+        for (size_t i = 0; i < sz; i++) {
+            subtable->set_int(0, i, value_list[i]);
+        }
+    };
+    set_list(table1->get_subtable(2, 0), std::vector<Int>({100, 200, 300, 400, 500}));
+    set_list(table1->get_subtable(2, 1), std::vector<Int>({1, 2, 3}));
+    set_list(table1->get_subtable(2, 2), std::vector<Int>({1, 2, 3, 4, 5}));
+    set_list(table1->get_subtable(2, 3), std::vector<Int>({1, 2, 3, 4, 5, 6, 7, 8, 9}));
+
+    auto set_links = [](LinkViewRef lv, const std::vector<int64_t>& value_list) {
+        for (auto v : value_list) {
+            lv->add(v);
+        }
+    };
+    set_links(table1->get_linklist(3, 0), std::vector<Int>({0, 1, 2, 3, 4, 5}));
+    set_links(table1->get_linklist(3, 1), std::vector<Int>({6, 7, 8, 9}));
+
+    Query q;
+    Query q1;
+    size_t match;
+    TableView tv;
+
+    q = strings.size() == 5;
+    q1 = table1->where().size_equal(0, 5);
+    match = q.find();
+    CHECK_EQUAL(1, match);
+    match = q1.find();
+    CHECK_EQUAL(1, match);
+
+    // Here the null values should not be included in the search
```

Comment:
The test here checks that the binary entries that are null, are not included in the search result. Was that what you meant?